src/cmd_line.rs

```diff
@@ -0,0 +1,72 @@
+use crate::settings::*;
+
+#[derive(Clone)]
+pub struct CmdLineSettings {
+    pub verbosity: u64,
+    pub log_to_file: bool,
+    pub neovim_args: Vec<String>,
+    pub files_to_open: Vec<String>,
+
+    pub disowned: bool,
+    pub geometry: Option<String>,
+    pub wsl: bool,
+    pub remote_tcp: Option<String>,
+    pub multi_grid: bool,
+    pub maximized: bool,
+}
+
+impl Default for CmdLineSettings {
+    fn default() -> Self {
+        Self {
+            verbosity: 0,
+            log_to_file: false,
+            neovim_args: vec![],
+            files_to_open: vec![],
+            disowned: false,
+            geometry: None,
+            wsl: false,
+            remote_tcp: None,
+            multi_grid: false,
+            maximized: false,
+        }
+    }
+}
+
+pub fn handle_command_line_arguments() {
+    let clapp = clap_app!(neovide =>
+    (author: crate_authors!())
+    (version: crate_version!())
+    (about: crate_description!())
+    (@arg verbosity: -v ... "Set the level of output information")
+    (@arg log_to_file: --log "Log to a file")
+    (@arg disowned: --disowned "Disown the process. (only on macos)")
+    (@arg maximized: --maximized "Maximize the window.")
+    (@arg multi_grid: --multi-grid "Enable Multigrid")
+    (@arg wsl: --wsl "Run in WSL")
+    (@arg remote_tcp: --remote-tcp +takes_value "Connect to Remote TCP")
+    (@arg geometry: --geometry +takes_value "Specify the Geometry of the window")
+
+    (@arg files: +takes_value +multiple "Open Files")
+    (@arg neovim_args: +takes_value +multiple +last "Args to pass to Neovim")
+                        );
+    let matches = clapp.get_matches();
+
+    SETTINGS.set::<CmdLineSettings>(&CmdLineSettings {
+        neovim_args: matches
+            .values_of("neovim_args")
+            .map(|opt| opt.map(|v| v.to_owned()).collect())
+            .unwrap_or_default(),
+        verbosity: matches.occurrences_of("verbosity"),
+        log_to_file: matches.is_present("log_to_file"),
+        files_to_open: matches
+            .values_of("files")
+            .map(|opt| opt.map(|v| v.to_owned()).collect())
+            .unwrap_or_default(),
+        maximized: matches.is_present("maximized"),
+        multi_grid: matches.is_present("multi_grid"),
```

Comment:
multigrid also sets via an environment variable. Is that handled here or where it is used?

Reply:
I did think of also refactoring the environment variables but really wanted to get your opinion on how to integrate them. Should they just be used as "defaults" for command-line arguments? And if yes should every cli-argument be replaceable through an environment variable? 

For now I simply didn't touch them, so it's still handled in [`src/bridge/mod.rs` ](https://github.com/ErikMayrhofer/neovide/blob/778f4b8cbcf501b1609950d163d4f771a45278c4/src/bridge/mod.rs#L273)

Reply:
I think a strategy where some environment variables are defined as defaults is a good way to go. That way its just handled in one place rather than specialized handling wherever its needed. What do you think?

Reply:
Yup, that seems like a reasonable way.

Just so that I make sure we are on the same page: 
* environment variables exist for some parameters
* environment variables are overwritten by command line parameters so `NeovideMultiGrid=false ./neovide --multiGrid` would result in `CmdLineSettings.multiGrid` being `true`
* this whole logic is handled in cmd_line.rs or a similar place and included in this PR in its final state

Reply:
Yes. That sounds perfect